src/plugins/chart_expressions/expression_partition_vis/public/utils/get_legend_actions.tsx

```diff
@@ -44,9 +47,17 @@ export const getLegendActions = (
     let formattedTitle = '';
     if (visParams.dimensions.buckets) {
       const column = visParams.dimensions.buckets.find(
-        (bucket) => bucket.accessor === filterData.data.data[0].column
+        (bucket) =>
+          (typeof bucket === 'string' ? bucket : bucket.accessor) === filterData.data.data[0].column
```

Comment:
Don't you want to move this comparison to the util function? It seems to me, this pattern is very frequent at this code.

Reply:
Yes I think I agree with @Kunzetsov here. It seems that we are repeating this check a lot.

Reply:
done, added new util method - `getAccessor`